Tests/xSharePoint/xSharePoint.xSPOutgoingEmailSettings.Tests.ps1

```diff
@@ -0,0 +1,101 @@
+[CmdletBinding()]
+param(
+    [string] $SharePointCmdletModule = (Join-Path $PSScriptRoot "..\Stubs\SharePoint\15.0.4693.1000\Microsoft.SharePoint.PowerShell.psm1" -Resolve)
+)
+
+$ErrorActionPreference = 'stop'
+Set-StrictMode -Version latest
+
+$RepoRoot = (Resolve-Path $PSScriptRoot\..\..).Path
+$Global:CurrentSharePointStubModule = $SharePointCmdletModule
+
+$ModuleName = "MSFT_xSPOutgoingEmailSettings"
+Import-Module (Join-Path $RepoRoot "Modules\xSharePoint\DSCResources\$ModuleName\$ModuleName.psm1")
+
+Describe "xSPOutgoingEmailSettings" {
+    InModuleScope $ModuleName {
+        $testParams = @{
+            WebAppUrl = "http://sp2013.australiasoutheast.cloudapp.azure.com"
+            SMTPServer = "smtp.contoso.com"
+            FromAddress = "from@email.com"
+            ReplyToAddress = "reply@email.com"
+            CharacterSet= "65001"
+        }
+        Import-Module (Join-Path ((Resolve-Path $PSScriptRoot\..\..).Path) "Modules\xSharePoint")
+        
+        Mock Invoke-xSharePointCommand { 
+            return Invoke-Command -ScriptBlock $ScriptBlock -ArgumentList $Arguments -NoNewScope
+        }
+        
+        Import-Module $Global:CurrentSharePointStubModule -WarningAction SilentlyContinue         
+
+
+
+        Context " Web Application isn't available " {
+            Mock Get-SPWebApplication -MockWith  { return $null
+            }
+
+            It "returns null from the get method" {
+                Get-TargetResource @testParams | Should be $null 
+            }
+
+            It "returns false from the test method" {
+                Test-TargetResource @testParams | Should be $false
+            }
+
+        }
+
+        Context " Properties match" {
+            Mock Get-SPWebApplication { 
+                return @{
+                        Url= "http://sp2013.australiasoutheast.cloudapp.azure.com"
```

Comment:
Can we make the URLs we mock with more generic examples, like http://sharepoint.contoso.com or something like that?